File: seminar_2/task_4/task4.cpp
#include <iostream>
#include <algorithm>

int copyTime(int numberCopy, int firstPrintner, int secondPrinter) {
    int left = 0; 
    int right = (numberCopy -1) * std::max(firstPrintner, secondPrinter);
    int middle = 0;
    
    while (left + 1 < right) {
        middle = (right + left) / 2;
        if (middle/firstPrintner + middle/secondPrinter < numberCopy - 1) {
            left = middle;
        } else {
            right = middle;
        }
    }
    
    return right + std::min(firstPrintner, secondPrinter);
}



int main() {
    size_t numberCopy, firstRinter, secondPrinter;
    std::cin >> numberCopy >> firstRinter >> secondPrinter;
    std::cout << copyTime(numberCopy, firstRinter, secondPrinter) << std::endl;
}

File: seminar_2/task_4/task_4.py
# очень простая задача (последняя)

def copy_time(n, x, y):
    
    l = 0 # самый удобный случай
    r = (n - 1) * max(x, y) # рассматриваем худший случай реализации

    while l + 1 < r:
        
        mid = (l + r) // 2
        
        if (mid // x) + (mid // y) < (n - 1):
            l = mid
        else:
            r = mid
            
        print(l, r)
        
    return r + min(x, y)    
            
    
print(copy_time(8, 1, 2))    
